### ISSUE TITLE
Cannot find module 'webpack' on fresh installation

### ISSUE THREAD

**Issue by nestorvanz** [nrwl/nx#5949]
I have a fresh installation of NX for React project; I got this message when try to serve the React project from VSCode NX toolkit.

```bash
> Executing task: nx serve fe-ui-compoents <


> nx run fe-ui-compoents:serve 
Cannot find module 'webpack'
Require stack:
- /Users/nv/projects/x/y/node_modules/webpack-dev-server/lib/Server.js
- /Users/nv/projects/x/y/node_modules/@nrwl/web/src/executors/dev-server/dev-server.impl.js
- /Users/nv/projects/x/y/node_modules/@nrwl/tao/src/shared/workspace.js
- /Users/nv/projects/x/y/node_modules/@nrwl/tao/src/commands/run.js
- /Users/nv/projects/x/y/node_modules/@nrwl/tao/index.js
- /Users/nv/projects/x/y/node_modules/@nrwl/cli/lib/run-cli.js

———————————————————————————————————————————————

>  NX   ERROR  Running target "fe-ui-compoents:serve" failed

  Failed tasks:
  
  - fe-ui-compoents:serve
```

**Comment by smasala** [comment]
Are you using Webpack 5?

**Comment by FrozenPandaz** [comment]
Was this created with `create-nx-workspace`? Which package manager was used? Please provide a repo if possible to help reproduce the issue.

**Comment by nestorvanz** [comment]
Steps followed
* Create a new project empty project
* Install React plugin to the project
* Create  new react project
* Serve the react project

Then shows the Webpack error.

I realized that Webpack is not getting installed, even is not appearing in the package.json.

I've tried to install Webpack by my own, and now I am getting the error listed on this [issue](https://github.com/nrwl/nx/issues/5907).

**Comment by dariusj18** [comment]
Same issue, deleted my `node_modules` folder and `package-lock.json` file and ran `npm install` and received same error.  I got the "tap" error when I installed webpack 5.39.0.  When I try and install webpack 4.42.0 I get

```
npm install -D webpack@4.42.0
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: type-fest@0.21.3
npm ERR! node_modules/type-fest
npm ERR!   type-fest@"^0.21.3" from ansi-escapes@4.3.2
npm ERR!   node_modules/ansi-escapes
npm ERR!     ansi-escapes@"^4.2.1" from @jest/core@26.6.3
npm ERR!     node_modules/@jest/core
npm ERR!       @jest/core@"^26.2.2" from jest@26.2.2
npm ERR!       node_modules/jest
npm ERR!         peer jest@">=26 <27" from ts-jest@26.5.5
npm ERR!         node_modules/ts-jest
npm ERR!         1 more (the root project)
npm ERR!       1 more (jest-cli)
npm ERR!     ansi-escapes@"^4.2.1" from jest-watcher@26.6.2
npm ERR!     node_modules/jest-watcher
npm ERR!       jest-watcher@"^26.6.2" from @jest/core@26.6.3
npm ERR!       node_modules/@jest/core
npm ERR!         @jest/core@"^26.2.2" from jest@26.2.2
npm ERR!         node_modules/jest
npm ERR!         1 more (jest-cli)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional type-fest@"^0.13.1" from @pmmmwh/react-refresh-webpack-plugin@0.4.3
npm ERR! node_modules/@pmmmwh/react-refresh-webpack-plugin
npm ERR!   @pmmmwh/react-refresh-webpack-plugin@"^0.4.3" from @nrwl/react@12.3.6
npm ERR!     dev @nrwl/react@"12.3.6" from the root project
```

I brought back my `package-lock.json` file and ran `npm install` again and web project worked again.

I had created the project via `npx create-nx-workspace`

Nodejs: v14.16.0  
npm: v7.16.0
nx: v12.3.6

**Comment by dariusj18** [comment]
This is marked as `blocked: more info needed` what more information do you need?

**Comment by dariusj18** [comment]
This is marked as blocked: more info needed what more information do you need?

**Comment by wwwsolutions** [comment]
### Same problem after upgraded to Nx 12+ 

I was expecting that this problem would have been fixed hence I would like to contribute to solution of this annoying bug.
I noticed it after upgrading to Nx 12 and it is easily reproducable.

___Problem happens when adding @nrwl/react to the empty workspace generated by create-nx-workspace command.___

Win 10 Pro
Nodejs: v15.11.0
npm: v7.15.0
nx: v12.5.8

1) Sequence with error:

    - npx create-nx-workspace
    - (name) > xxx
    - (preset) > empty
    - (nx cloud) > no
    - npm i -D @nrwl/react
    - nx generate @nrwl/react:application --name=test-app
    - nx serve
    - ERROR

![WindowsTerminal_2021-07-15_16-00-10](https://user-images.githubusercontent.com/42439126/125800813-a4eff081-f17f-4267-abb3-ca696b163ca6.png)

2. Sequence without error:

    - npx create-nx-workspace
    - (name) > yyy
    - (preset) > react
    - (app name) > test-app
    - (nx cloud) > no
    - nx serve
    - SUCCESS

![WindowsTerminal_2021-07-15_16-10-19](https://user-images.githubusercontent.com/42439126/125802322-1fee7cf8-10d4-4029-92a4-746615a349ea.png)

**Comment by nestorvanz** [comment]
What is the current state of this bug? Keeps happening for new empty projects.

**Comment by dariusj18** [comment]
This is marked as blocked: more info needed what more information do you need?

**Comment by dariusj18** [comment]
This is marked as blocked: more info needed what more information do you need?

**Comment by dariusj18** [comment]
This is marked as blocked: more info needed what more information do you need?

**Comment by npwork** [comment]
Reproduces.... :(

**Comment by CaioRolla** [comment]
Same thing here after upgrade.

Cannot find module 'webpack'
Require stack:
- /Users/caiorolla/projects/nui/node_modules/@angular-devkit/build-webpack/src/webpack/index.js
- /Users/caiorolla/projects/nui/node_modules/@angular-devkit/build-webpack/src/index.js
- /Users/caiorolla/projects/nui/node_modules/@angular-devkit/build-angular/src/dev-server/index.js
- /Users/caiorolla/projects/nui/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /Users/caiorolla/projects/nui/node_modules/@angular-devkit/architect/node/index.js
- /Users/caiorolla/projects/nui/node_modules/@nrwl/tao/src/commands/ngcli-adapter.js
- /Users/caiorolla/projects/nui/node_modules/@nrwl/tao/src/commands/run.js
- /Users/caiorolla/projects/nui/node_modules/@nrwl/tao/index.js
- /Users/caiorolla/projects/nui/node_modules/@nrwl/cli/lib/run-cli.js

**Comment by SzymonPawlak** [comment]
We have exactly the same issue with version 11

**Comment by fabiano-amaral** [comment]
Same problem here after a fresh installation, using angular project, solved with:

```cli
npm install --save-dev webpack

npm install --save-dev webpack-dev-server
```

**Comment by ngfelixl** [comment]
I faced the same problem with
- `npx create-nx-workspace` -> preset: empty
- `npm i --save-dev @nrwl/angular`
- `npx nx g @nrwl/angular:application` -> name: app
- `npx nx run app:build`

Installed nx version is 13.3.1

Resulting in 
```
Cannot find module 'webpack'
Require stack:
- C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-webpack\src\webpack\index.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-webpack\src\index.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-angular\src\builders\browser\index.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\architect\node\node-modules-architect-host.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\architect\node\index.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\src\commands\ngcli-adapter.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\src\commands\run.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\index.js
- C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\cli\lib\run-cli.js
```

Resolved it with 
```
npm i --save-dev webpack
```

**Comment by Eskobastion** [comment]
> I faced the same problem with
> 
> * `npx create-nx-workspace` -> preset: empty
> * `npm i --save-dev @nrwl/angular`
> * `npx nx g @nrwl/angular:application` -> name: app
> * `npx nx run app:build`
> 
> Installed nx version is 13.3.1
> 
> Resulting in
> 
> ```
> Cannot find module 'webpack'
> Require stack:
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-webpack\src\webpack\index.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-webpack\src\index.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\build-angular\src\builders\browser\index.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\architect\node\node-modules-architect-host.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@angular-devkit\architect\node\index.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\src\commands\ngcli-adapter.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\src\commands\run.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\tao\index.js
> - C:\Users\felix\Documents\development\felixlemke\node_modules\@nrwl\cli\lib\run-cli.js
> ```
> 
> Resolved the build and serve target issue by installing webpack manually:
> 
> ```
> npm i --save-dev webpack
> ```

I had exactly the same problem with same configuration today. Installing webpack resolved it.

**Comment by sneerin** [comment]
not sure just use the manual , created empty workspace run the same command, and it failed. such a shame

**Comment by wojtek-viirtue** [comment]
This is still happening for 13.8.0, @fabiano-amaral solution solves it.

**Comment by heynitish717** [comment]
nx version - 13.8.0
@nrwl/angular plugin for nx version - 13.8.0
This issue is still happening on choosing empty workspace 
running "npm i --save-dev webpack" resolves the issue.

@vsavkin sir please get below sequence of steps updated for empty workspace option on nx official documentation. Lot of us are getting this issue

```
1.npx create-nx-workspace@latest
--Enter workspace name - demo-app2
--What to create in the new workspace - apps (creates empty workspace)
--Use Nx Cloud? No

Warning - use "npx nx" or run "npm install -g nx"

2.npm install -g nx
3.code demo-app2 (opens folder in VSCode)
4.npm install @nrwl/angular
5.ng g @nrwl/angular:app customer-portal --routing
6.nx serve customer-portal
Error: Cannot find module 'webpack'

7.npm i --save-dev webpack

8.nx serve customer-portal
Success -  open your browser on http://localhost:4200/**
```

**Comment by heynitish717** [comment]
> 

Yes I too got the same issue and had to install webpack manually using below command:
```npm i --save-dev webpack```
It resolved my issue

**Comment by JWess** [comment]
I got this error because I had two dependencies (`webpack-dev-server` and `@types/webpack`) that required different versions of `webpack` as their own dependency. This caused `webpack` to be installed in the subdirectories of those dependencies instead of directly at the root of the `node_modules` directory (i.e. `node_modules/webpack`). Once I removed `@types/webpack` from my `devDependencies`, deleted my `package-lock.json` and `node_modules`, and ran `npm i`, the issue went away.
